### DISCUSSION TITLE
Privacy and information integrity

### DISCUSSION THREAD

**Discussion by paviad** [mangate/SelfGPT#16] [Ideas]
How do we approach the issue of privacy? Since this is a whatsapp bot with a phone number, everyone who has the phone number can send it a message.

There are a few options as I see it:

1. Do nothing, just keep the number a secret (not very feasible in the long term)
2. Configure your bot to look at the number of the sender and if it is unauthorized, simply reject all messages
3. Configure the bot to handle a set of authorized senders with a common database e.g. a married couple or some partnership
4. Configure the bot to use a separate database for each sender
5. Mix all of the above, associate every sender with a database id, multiple senders can be associated with the same id. Unassociated senders are rejected.

Ideas?

**Comment by evbambly** [comment]
Twilio manages numbers which can contact a given Twilio number and do not allow random connections.
IMO At this scale of the project, we can use this feature and be content with it.
Additionally, this allows seamless sharing of databases between users which is a plus.

<img width="302" alt="Screenshot 2023-02-19 153646" src="https://user-images.githubusercontent.com/45696895/219951768-92eb9d9d-3ce1-4da8-9f20-32deaf2dd494.png">

<img width="651" alt="Screenshot 2023-02-19 153453" src="https://user-images.githubusercontent.com/45696895/219951634-2e26d43b-7d10-4a92-ba86-9483dc6f4b14.png">
